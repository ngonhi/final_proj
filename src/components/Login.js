import React, { Component } from 'react'
import {Redirect} from 'react-router-dom'

class Login extends Component {
  constructor() {
    super()
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit(event) {
    console.log(this.props)
    event.preventDefault()

    const user = {
        "username": event.target.elements.username.value,
        "password": event.target.elements.password.value
    }

    fetch("http://127.0.0.1:5000/login", {
      method: 'POST',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
        },
      body: JSON.stringify(user)
    })
    .then(res => res.json())
    .then(data => {
      this.props.insertToken(data)
      this.props.onHistory.push(`/Categories`)
    })
    .catch(error => console.log(error.message))
}
    

  render() {
    return (
        <div className='form'>
                <form onSubmit={this.handleSubmit}> 
                    <input type='text' placeholder='username' name='username'></input>
                    <input type='text' placeholder='password' name='password'></input>
                    <button> Login </button>
                </form>
            </div>
    )
  }
}

export default Login