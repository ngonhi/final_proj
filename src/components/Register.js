import React, { Component } from 'react'

class Register extends Component {
  constructor() {
    super()
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit(event) {
    event.preventDefault()

    const newUser = {
        "email": event.target.elements.email.value,
        "username": event.target.elements.username.value,
        "password": event.target.elements.password.value,
        "name": event.target.elements.name.value
    }

    console.log(newUser)
    fetch("http://127.0.0.1:5000/registrations", {
      method: 'POST',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
        },
      body: JSON.stringify(newUser)
    })
    .then(res => res.json())
    .then(data => console.log(data)) // Access token
    .catch((error) => console.log(error.message)) // Error message. Use alert

    // Insert users into state
    this.props.history.push(`/Categories`)
  }

  render() {
    return (
        <div className='form'>
                <form onSubmit={this.handleSubmit}> 
                    <input type='text' placeholder='username' name='username'></input>
                    <input type='text' placeholder='name' name='name'></input>
                    <input type='text' placeholder='email' name='email'></input>
                    <input type='text' placeholder='password' name='password'></input>
                    <button> Register </button>
                </form>
            </div>
    )
  }
}

export default Register