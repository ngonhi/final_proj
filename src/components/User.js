import React from 'react'
import {Link} from 'react-router-dom'

function User(props) {
    return (
        <div className='button-container'>
            <Link to='/Register' className='button'> Register </Link>
            <Link to='/Login' className='button'> Login </Link>
        </div>
    )
}

export default User