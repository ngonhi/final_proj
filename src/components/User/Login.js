import React, { Component } from 'react'

class Login extends Component {
  componentDidUpdate() {
    if (this.props.access_token && Object.keys(this.props.user).length === 0) {
    const url = window.$domain + '/me' 
      const option = {
        method: 'GET',
        headers: {
          'Authorization': 'Bearer ' + this.props.access_token,
          'Content-Type': 'application/json'
        }
      }
    this.props.fetchRequestObj("START_LOADING_USER", url, option)
    }
  }

  handleSubmit = (event) => {
    this.props.clearError()
    event.preventDefault()
    const {username, password} = event.target.elements
    const user = {
        "username": username.value,
        "password": password.value
    }

    const url = window.$domain + '/login'
    const option = {
      method: 'POST',
      headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json'
      },
      body: JSON.stringify(user)
    }

    this.props.fetchRequestObj("START_LOADING_TOKEN", url, option)
    .then(() => {
      if(Object.keys(this.props.error).length !== 0) {
        this.props.history.push('/login')
      } else {
      this.props.history.push('/categories')
      }
    })
  }

  handleError = (error) => {
    if (Object.keys(error).length !== 0) {
      const {message, status, statusText} = error
      error = <div className='error'> {status} - {statusText} - {message} </div>
    } else {
      error = null
    }

    return error
  }
  
  render() {
    let error = this.props.error
    if (error) { error = this.handleError(error) }

    return (
      <div>
        <div className='form'>
                <form onSubmit={this.handleSubmit}> 
                    <input type='text' placeholder='username' name='username'></input>
                    <input type='text' placeholder='password' name='password'></input>
                    <button> Login </button>
                </form>
        </div>
        {error}
      </div>
    )
  }
}

export default Login