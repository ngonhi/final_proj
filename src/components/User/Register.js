import React, { Component } from 'react'

class Register extends Component {
  componentDidUpdate() {
    if (this.props.access_token && Object.keys(this.props.user).length === 0) {
    const url = window.$domain + '/me' 
      const option = {
        method: 'GET',
        headers: {
          'Authorization': 'Bearer ' + this.props.access_token,
          'Content-Type': 'application/json'
        }
      }
    this.props.fetchRequestObj("START_LOADING_USER", url, option)
    }
  }

  handleSubmit = (event) => {
    this.props.clearError()
    event.preventDefault()
    const {username, name, email, password} = event.target.elements
    const newUser = {
        "username": username.value,
        "name": name.value,
        "email": email.value,
        "password": password.value
    }

    const url = window.$domain + '/registrations'
    const option = {
      method: 'POST',
      headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json'
      },
      body: JSON.stringify(newUser)
    }
    this.props.fetchRequestObj("START_LOADING_TOKEN", url, option)
    .then(() => {
      if(Object.keys(this.props.error).length !== 0) {
        this.props.history.push('/register')
      } else {
      this.props.history.push('/categories')
      }
    })
  }

  handleError = (error) => {
    if (Object.keys(error).length !== 0) {
      const {message, status, statusText} = error
      const mess_list = Object.values(message)
      error = <div className='error'> {status} - {statusText} - {mess_list.join(' ')} </div>
    } else {
      error = null
    }

    return error
  }

  render() {
    let error = this.props.error
    if (error) { error = this.handleError(error) }

    return (
      <div>
        <div className='form'>
                <form onSubmit={this.handleSubmit}> 
                    <input type='text' placeholder='username' name='username'></input>
                    <input type='text' placeholder='name' name='name'></input>
                    <input type='text' placeholder='email' name='email'></input>
                    <input type='text' placeholder='password' name='password'></input>
                    <button> Register </button>
                </form>
        </div>
        {error}
      </div>
    )
  }
}

export default Register