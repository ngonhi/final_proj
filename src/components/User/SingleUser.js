import React from 'react'
import NavBar from '../NavBar'

function SingleUser(props) {
    const user = props.user
    return (
        <div>
            <title> User </title>
            <NavBar {...props}/>
            <center>
            <p><b>Name:</b> {user.name}</p>
            <p><b>Username:</b> {user.username}</p>
            <p><b>Email:</b> {user.email}</p>
            </center>
        </div>
    )
}

export default SingleUser