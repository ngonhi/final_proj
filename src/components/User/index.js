import React from 'react'
import {Link} from 'react-router-dom'

function User(props) {
    if (props.error) {
    if (Object.keys(props.error).length !== 0) {
        props.clearError()
    }
}
    
    return (
        <div className='button-container'>
            <Link to='/register' className='button'> Register </Link>
            <Link to='/login' className='button'> Login </Link>
        </div>
    )
}

export default User