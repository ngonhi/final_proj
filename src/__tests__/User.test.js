import React from 'react';
import { Link } from 'react-router-dom';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Register from '../components/User/Register';
import Login from '../components/User/Login';

import User from '../components/User/index';

configure({ adapter: new Adapter() });

const props = {
  fetchRequestObj: jest.fn().mockResolvedValue({ payload: { result: {} } }),
  error: [],
  clearError: jest.fn(),
};

describe('<User />', () => {
  const wrapper = shallow(<User />);
  it('should contain button-container element', () => {
    expect(wrapper.find('.button-container').exists()).toBeTruthy();
  });
  it('button-container should contain two children', () => {
    expect(wrapper.find('.button-container').children().length).toBe(2);
  });
  it('should have proper snapshot', () => {
    const wrapper = shallow(<User />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('link should render correctly', () => {
    const wrapper = shallow(<User />);
    expect(wrapper.find(Link).at(0).props().to).toBe('/register');
    expect(wrapper.find(Link).at(1).props().to).toBe('/login');
  });
});


describe('<Login /> rendering', () => {
  it('should have proper snapshot', () => {
    const wrapper = shallow(<Login />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call fetchRequestObj when submit', () => {
    const wrapper = mount(<Login {...props} />);
    wrapper.find('form').simulate('submit');
    const inputEle = wrapper.find('input');
    const user = {
      username: inputEle.at(0).instance().value,
      password: inputEle.at(1).instance().value,
    };

    const url = `${window.$domain}/login`;
    const option = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(user),
    };
    const type = 'START_LOADING_TOKEN';
    expect(props.fetchRequestObj).toHaveBeenCalledWith(
      type, url, option,
    );
    wrapper.unmount();
  });
  it('should contain .form class', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper.find('.form').exists()).toBeTruthy();
  });
});


describe('<Register /> rendering', () => {
  it('should have proper snapshot', () => {
    const wrapper = shallow(<Register />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call fetchRequestObj when submit', () => {
    const wrapper = mount(<Register {...props} />);
    wrapper.find('form').simulate('submit');

    const inputEle = wrapper.find('input');
    const newUser = {
      username: inputEle.at(0).instance().value,
      name: inputEle.at(1).instance().value,
      email: inputEle.at(2).instance().value,
      password: inputEle.at(3).instance().value,
    };
    const url = `${window.$domain}/registrations`;
    const option = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(newUser),
    };
    const type = 'START_LOADING_TOKEN';
    expect(props.fetchRequestObj).toHaveBeenCalledWith(
      type, url, option,
    );
    wrapper.unmount();
  });
  it('should contain .form class', () => {
    const wrapper = shallow(<Register />);
    expect(wrapper.find('.form').exists()).toBeTruthy();
  });
});
