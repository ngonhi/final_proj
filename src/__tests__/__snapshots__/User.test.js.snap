// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> rendering should have proper snapshot 1`] = `
<div>
  <div
    className="form"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        name="username"
        placeholder="username"
        type="text"
      />
      <input
        name="password"
        placeholder="password"
        type="text"
      />
      <button>
         Login 
      </button>
    </form>
  </div>
</div>
`;

exports[`<Register /> rendering should have proper snapshot 1`] = `
<div>
  <div
    className="form"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        name="username"
        placeholder="username"
        type="text"
      />
      <input
        name="name"
        placeholder="name"
        type="text"
      />
      <input
        name="email"
        placeholder="email"
        type="text"
      />
      <input
        name="password"
        placeholder="password"
        type="text"
      />
      <button>
         Register 
      </button>
    </form>
  </div>
</div>
`;

exports[`<User /> should have proper snapshot 1`] = `
<div
  className="button-container"
>
  <Link
    className="button"
    to="/register"
  >
     Register 
  </Link>
  <Link
    className="button"
    to="/login"
  >
     Login 
  </Link>
</div>
`;
